client-lite/src/exe/docs.cpp

```diff
@@ -16,57 +16,95 @@ namespace msdoutil = microsoft::deliveryoptimization::util::details;
 
 #include "config_manager.h"
 #include "download_manager.h"
+#include "do_noncopyable.h"
 #include "proc_launch_helper.h"
 #include "rest_http_controller.h"
 #include "rest_port_advertiser.h"
 #include "trace_sink.h"
 
 using namespace std::chrono_literals; // NOLINT(build/namespaces) how else should we use chrono literals?
 
-class ProcessController
+class ProcessController : DONonCopyable
 {
 public:
     ProcessController()
     {
-        // Do init work like registering signal control handler
-        signal(SIGINT, _SignalHandler);
-        signal(SIGTERM, _SignalHandler);
     }
 
-    void WaitForShutdown(const std::function<bool()>& fnIsIdle)
+    void SetSignalHandler(const std::function<void()>& fnHandler, int signalNumber)
     {
+        if (signalNumber == SIGINT)
+        {
+            _sigintHandler = std::move(fnHandler);
+        }
+        else if (signalNumber == SIGTERM)
+        {
+            _sigtermHandler = std::move(fnHandler);
+        }
+        else if (signalNumber == SIGHUP)
+        {
+            _sighupHandler = std::move(fnHandler);
+        }
```

Comment:
Not sure why but a switch case doesn't properly map the int signal input to the actual signal value. i.e. something like this wasn't working

switch (signal)
{
case SIGINT:
...
case SIGTERM:
...
case SIGHUP:
...
}